cmd/kube-scheduler/app/options/options.go

```diff
@@ -78,26 +77,11 @@ func NewOptions() (*Options, error) {
 		return nil, err
 	}
 
-	hhost, hport, err := splitHostIntPort(cfg.HealthzBindAddress)
```

Comment:
I don't see this field being used anywhere after this change. Shouldn't it be used for the secure address?

Reply:
Secure serving has its own flags added through https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/serving.go#L147-L215

Reply:
So now this field is useless. Do we want to keep it like that? Maybe we can make it influence the secure address starting v1beta2. cc @adtac

Reply:
What is your answer to this @ingvagabund ?

Reply:
I was under impression you are cc'ing @adtac to answer the question.

The same holds for `MetricsBindAddress`.

> Maybe we can make it influence the secure address starting v1beta2

That would require to either run two secure servers or extend the current secure server code to allow two different addresses to be listened to. What would be a benefit of two secure addresses instead of one?

Reply:
I CCed Adi for awareness, as he is working in CC v1beta2

What I'm saying is that the field should stop determining the insecure address and determine the secure one. Or, we just remove the field.

Reply:
+1 for removing both fields.

Reply:
any other thoughts @ahg-g @Huang-Wei?
Should it be possible to configure the secure port via CC?

Reply:
> Should it be possible to configure the secure port via CC?

It is probably better to follow an approach common with other components, what do other components do?

Reply:
Sorry guys I don't have sufficient knowledge in this area to give an educated answer. I can't talk for other components. Checking the codebase `kube-proxy`, `kubeadm` and `kubelet` utilize `HealthzBindAddress`. `MetricsBindAddress` is utilized only by `kube-proxy`. I don't see any use in KCM.

Reply:
I don't see `kube-proxy`, `kubeadm` or `kubelet` relying on `k8s.io/apiserver/pkg/server.SecureServing`. Which is the reason why neither of `HealthzBindAddress` or `MetricsBindAddress` is no longer read.

Reply:
I'm fine with removing the fields in the internal type and v1beta2 (now merged)

Reply:
removing sounds good to me.

Reply:
Done. Do we need to mention it in the release notes?

Reply:
We need to mention that as of v1beta1 the fields are a no-op and that they are removed as of v1beta2.

Reply:
we can keep fields in the internal type until v1beta1 is removed, and in validation we validate that the two strings are empty. This is similar to `port=0` flag validation.

Reply:
Added validation to check both fields has to be empty.